### ISSUE TITLE
no-undef triggers false positive with es6 classes

### ISSUE THREAD

**Issue by rosshadden** [eslint/eslint#3941]
When using `no-undef`, references to classes result in an error.  The following is an example with three common uses that trigger this false positive.

```javascript
class Foo {
  static create() {
    return new Foo('blah');
    // "Foo" is not defined. (no-undef)
  }
}

let foo = new Foo();
// "Foo" is not defined. (no-undef)

Foo.create();
// "Foo" is not defined. (no-undef)
```

This occurs in both eslint 1.3.1 and 1.5.1.

/cc @crispy1989 @oppenlander

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
I think the rule doesnt account for classes. I might be wrong too.
Need someone else to verify.

**Comment by nzakas** [comment]
This might be an issue with escope, since this rule relies on scope info almost exclusively.

**Comment by mysticatea** [comment]
Hmm?
I might not understand this issue, but, I cannot reproduce this in the online demo.

![image](https://cloud.githubusercontent.com/assets/1937871/10123513/5e082a66-6576-11e5-8614-de737f45067e.png)

**Comment by nzakas** [comment]
@rosshadden can you provide your configuration?

**Comment by rosshadden** [comment]
Maybe I had a typo in my code which was causing it.  I'll close this untill/unless I can recreate a minimal example.  Thanks!
